src/ImageSharp/Formats/Tiff/TiffFrameMetadata.cs

```diff
@@ -0,0 +1,331 @@
+        /// <value>The density of the image in x- direction.</value>
+        public double? HorizontalResolution
+        {
+            get
+            {
+                if (this.ResolutionUnit != TiffResolutionUnit.None)
+                {
+                    double resolutionUnitFactor = this.ResolutionUnit == TiffResolutionUnit.Centimeter ? 2.54 : 1.0;
+
+                    if (this.TryGetSingle(ExifTag.XResolution, out Rational xResolution))
+                    {
+                        return xResolution.ToDouble() * resolutionUnitFactor;
+                    }
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Gets the resolution of the image in y- direction.
+        /// </summary>
+        /// <value>The density of the image in y- direction.</value>
+        public double? VerticalResolution
+        {
+            get
+            {
+                if (this.ResolutionUnit != TiffResolutionUnit.None)
+                {
+                    double resolutionUnitFactor = this.ResolutionUnit == TiffResolutionUnit.Centimeter ? 2.54 : 1.0;
+
+                    if (this.TryGetSingle(ExifTag.YResolution, out Rational yResolution))
+                    {
+                        return yResolution.ToDouble() * resolutionUnitFactor;
+                    }
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Gets how the components of each pixel are stored.
+        /// </summary>
+        public TiffPlanarConfiguration PlanarConfiguration => this.GetSingleEnum<TiffPlanarConfiguration, ushort>(ExifTag.PlanarConfiguration, DefaultPlanarConfiguration);
+
+        /// <summary>
+        /// Gets the unit of measurement for XResolution and YResolution.
+        /// </summary>
+        public TiffResolutionUnit ResolutionUnit => this.GetSingleEnum<TiffResolutionUnit, ushort>(ExifTag.ResolutionUnit, DefaultResolutionUnit);
+
+        /// <summary>
+        /// Gets the name and version number of the software package(s) used to create the image.
+        /// </summary>
+        public string Software => this.GetString(ExifTag.Software);
+
+        /// <summary>
+        /// Gets the date and time of image creation.
+        /// </summary>
+        public string DateTime => this.GetString(ExifTag.DateTime);
+
+        /// <summary>
+        /// Gets the person who created the image.
+        /// </summary>
+        public string Artist => this.GetString(ExifTag.Artist);
+
+        /// <summary>
+        /// Gets the computer and/or operating system in use at the time of image creation.
+        /// </summary>
+        public string HostComputer => this.GetString(ExifTag.HostComputer);
+
+        /// <summary>
+        /// Gets a color map for palette color images.
+        /// </summary>
+        public ushort[] ColorMap => this.GetArray<ushort>(ExifTag.ColorMap, true);
```

Comment:
I'm concerned about these being arrays. You can never update them without replacing the entire array.

If they are not designed to be updated, should they be public?

Reply:
@JimBobSquarePants Yes, I think this field can make as internal, currently, it use only for internals.
[ColorMap](https://www.awaresystems.be/imaging/tiff/tifftags/colormap.html) use for palette images, I think, it will not be edited element by element. it's possible to generate a palette with scan entire image when saving.